### DISCUSSION TITLE
crud image type field & path

### DISCUSSION THREAD

**Discussion by blondie63** [Laravel-Backpack/community-forum#905] [Q&A (Help)] [answered]
Following https://backpackforlaravel.com/docs/6.x/crud-fields#image-pro
i've create this crud field:
```
        CRUD::field([
            'name'          => 'icon',
            'label'         => trans('backpack::base.icon'),
            'type'          => 'image',
            'aspect_ratio'  => 1,
            'crop'          => true,
            'withFiles'     => true,
            'tab' => trans('backpack::base.attributes')
        ]);
```
I've default filesystem.php entry:
```
        'public' => [
            'driver' => 'local',
            'root' => storage_path('app/public'),
            'url' => env('APP_URL').'/storage',
            'visibility' => 'public',
            'throw' => false,
        ],
```
I've created the right symlink with cmd: php artisan storage:link
I add a new record and choose an image then save it
Now i see the new record in list but image preview is wrong, this is the link: http://localhost:8040/wzS39HXZkueEFF9bAje20WXzErDVndmVZwejJqaZ.png

Instead if i modify same record i see the right image preview that have this link:
http://localhost:8040/storage/wzS39HXZkueEFF9bAje20WXzErDVndmVZwejJqaZ.png

Looking my file system i see the uploaded file here:
$appPath/storage/app/public

Why image preview have a bad image link instead modify record have a right image link?
What's wrong ?

Thanks for help

**Comment by pxpm** [comment]
Sorry @blondie63 can you repost the screenshots with public urls ? I can't really access your localhost 😄

**Reply by blondie63** [reply]
This screenshots come from live dev server:
<img width="1386" alt="Screenshot 2024-03-29 alle 18 38 48" src="https://github.com/Laravel-Backpack/community-forum/assets/1261851/90fa0737-6e0c-4db9-8de8-2004b7761b68">
<img width="1374" alt="Screenshot 2024-03-29 alle 18 39 14" src="https://github.com/Laravel-Backpack/community-forum/assets/1261851/12687cf0-a14e-4860-a4fc-fe4055f1a028">

**Comment by karandatwani92** [comment]
Hey @pxpm 

It's a BUG; I'm also facing it.

[The Image Column Line:32](https://github.com/Laravel-Backpack/CRUD/blob/f2878a679ba74300aec69c9d0a13f020d7b87be4/src/resources/views/crud/columns/image.blade.php#L32)
`$href = $src = asset($column['prefix'] . $column['value']);`

The asset() helper doesn't append any default disk. It just appends the `ASSET_URL` env value.

To make it work I have to define the disk `CRUD::column('image')->type('image')->disk('public');`.

**Reply by blondie63** [reply]
ok I imagine it will be fixed in one of the next versions..

**Reply by blondie63** [reply]
@karandatwani92 I've tried to create a different disk in file system:
```
        'uploads' => [
            'driver' => 'local',
            'root' => storage_path('app/public/'),
            'url' => env('APP_URL').'/storage/uploads',
            'visibility' => 'public',
            'throw' => false,
        ],
```
        
        and use it in this CRUD:
```
        CRUD::field([
            'name'          => 'icon',
            'label'         => trans('backpack::base.icon'),
            'type'          => 'image',
            'disk'          => 'uploads',
            'aspect_ratio'  => 1,
            'crop'          => true,
            'withFiles'     => true,
            'tab' => trans('backpack::base.attributes')
        ]);
```

But there's the same problem..

**Reply by karandatwani92** [reply]
Hey @blondie63 
The disk should be in `withFiles`:
```php
        CRUD::field([
            'name'          => 'icon',
            'label'         => trans('backpack::base.icon'),
            'type'          => 'image',
            'aspect_ratio'  => 1,
            'crop'          => true,
            'withFiles' => [
                'disk' => 'public',
                'path' => 'uploads',
            ],
            'tab' => trans('backpack::base.attributes')
        ]);
```
AND
The custom disk `url` must have its symbolic link configured in the `links` array of `config/filesystems.php`

**Reply by blondie63** [reply]
@karandatwani92 with this setting file will be uploaded in the right folder and modifying record i see the image:
path is http://localhost:8040/storage/uploads/GQuagPlpVANVN7wDrkogd2ORJmG5tHhVJy5iSuGd.png
but image preview in list & show operation is wrong, path is:
http://localhost:8040/uploads/GQuagPlpVANVN7wDrkogd2ORJmG5tHhVJy5iSuGd.png

How is possible ? BKP bug ?

**Reply by karandatwani92** [reply, marked as answer]
@blondie63 You need to define the disk on the column. `CRUD::column('icon')->type('image')->disk('public');`

**Reply by blondie63** [reply]
Ok thanks, my mistake.. it working fine now!
